test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -828,6 +829,294 @@ def init_proc_mon(self):
+            default_vals = none_vals
+            cmds_to_exec = []
+            dmns_to_start = 0
+            pbs_conf_filepath = self.du.get_pbs_conf_file(server.hostname)
+            pbs_conf_val = self.du.parse_pbs_config(server.hostname,
+                                                    pbs_conf_filepath)
+            if not pbs_conf_val or len(pbs_conf_val) < 1:
+                raise ValueError("Could not parse pbs.conf on host %s" %
+                                 (server.hostname))
+
+            # to start with, set all keys in default_vals with values from the
+            # existing pbs.conf
+            for conf in default_vals:
+                try:
+                    default_vals[conf] = pbs_conf_val[conf]
+                except KeyError:
+                    # existing pbs.conf doesn't have a default variable set
+                    # Try to determine the default value for it
+                    val = self._get_dflt_pbsconfval(conf,
+                                                    primary_server.hostname,
+                                                    "server", server)
+                    if val is None:
+                        print "Couldn't revert %s in pbs.conf" % (key)
+                        raise
+                    else:
+                        default_vals[conf] = val
+
+            # Set all start bits
+            if (default_vals["PBS_START_SERVER"] != "1"):
+                default_vals["PBS_START_SERVER"] = "1"
+                dmns_to_start += 1
+                cmds_to_exec.append(["server", "start"])
+            if (default_vals["PBS_START_SCHED"] != "1"):
+                default_vals["PBS_START_SCHED"] = "1"
+                cmds_to_exec.append(["sched", "start"])
+                dmns_to_start += 1
+            if self.moms and server.hostname not in self.moms:
+                if default_vals["PBS_START_MOM"] != "0":
+                    default_vals["PBS_START_MOM"] = "0"
+                    cmds_to_exec.append(["mom", "stop"])
+            else:
+                if (default_vals["PBS_START_MOM"] != "1"):
+                    default_vals["PBS_START_MOM"] = "1"
+                    cmds_to_exec.append(["mom", "start"])
+                    dmns_to_start += 1
+            if self.comms and server.hostname not in self.comms:
+                if default_vals["PBS_START_COMM"] != "0":
+                    default_vals["PBS_START_COMM"] = "0"
+                    cmds_to_exec.append(["comm", "stop"])
+            else:
+                if (default_vals["PBS_START_COMM"] != "1"):
+                    default_vals["PBS_START_COMM"] = "1"
+                    cmds_to_exec.append(["comm", "start"])
+                    dmns_to_start += 1
+
+            # Set PBS_CORE_LIMIT, PBS_SCP and PBS_SERVER
+            if default_vals["PBS_CORE_LIMIT"] != "unlimited":
+                default_vals["PBS_CORE_LIMIT"] = "unlimited"
+                dmns_to_start = 4
```

Comment:
why are you setting this to 4 explicitly? what if a server do not have a mom on it?

Reply:
4 is just a way of expressing "restart all daemons on the server". I could have used another flag for this but I didn't think that it was needed. Please let me know if you think that I shouldn't overload dmns_to_start and use another flag instead.

Reply:
Actually i ended up refactoring it, please let me know if it's still confusing, thanks!

Reply:
Thanks @agrawalravi90. I looked at the updated  code and I am clear now :)